Docs: Update mount presign and .gitignore (#7887)
treeverse/lakeFS: docs/reference/mount.md

### Does data pass through the lakeFS server when using lakeFS Mount?

> lakeFS Mount leverages  pre-signed URLs to read data directly from the underlying object store, meaning data doesn’t  pass through the lakeFS server. By default, presign is enabled. To disable it, use:

Reviewer: didnt we agree that presigned is only enabled by default if supported by the underlying object store or false otherwise?

Author: @ozkatz The new behavior introduced in [PR](https://github.com/treeverse/fluffy/pull/290) is the same as `lakectl fs upload` exactly. 

`everest` will behave the following way:

- if `presign` is enabled by the server it **will use** `presign` otherwise if not support by the server it will not use `presign` 
- if user set explicitly `--presign=false` we will **not use** `presign`.

Ensure your **cache size** is large enough to accommodate the volume of files being accessed.

### How does lakeFS Mount integrate with a Git repository?

> It is perfectly safe to mount a lakeFS path within a Git repository, lakeFS Mount will add a virtual `.gitignore` file to the mounted directory to avoid the case of adding to the git repository the mounted objects from lakeFS (i.e running `git add -A`).

Reviewer 2: ```suggestion
It is perfectly safe to mount a lakeFS path within a Git repository.
lakeFS Mount prevents git from adding mounted objects to the git repository (i.e when running `git add -A`) by adding a virtual `.gitignore` file to the mounted directory.

```
